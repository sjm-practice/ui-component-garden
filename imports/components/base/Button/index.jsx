import React from "react";
import PropTypes from "prop-types";
import BaseButton from "./styles";

const Button = props => <BaseButton {...props} />;

Button.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
};

export default Button;
