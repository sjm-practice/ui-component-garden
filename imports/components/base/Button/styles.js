import styled from "styled-components";

const BaseButton = styled.button`
  background: transparent;
  border-radius: 3px;
  border: 2px solid palevioletred;
  color: palevioletred;
  margin: 0 1em;
  padding: 0.25em 1em;
  &:hover {
    border: 2px solid darkblue;
  }
`;

export default BaseButton;
