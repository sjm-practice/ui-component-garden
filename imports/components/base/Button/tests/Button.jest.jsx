/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import Button from "..";

describe("<Button />", () => {
  it("should match render snapshot", () => {
    const tree = renderer.create(<Button>Test</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
