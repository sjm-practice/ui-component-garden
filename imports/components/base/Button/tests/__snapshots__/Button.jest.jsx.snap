// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match render snapshot 1`] = `
<button
  className="sc-bdVaJa dMrfZP"
>
  Test
</button>
`;
